src/processes/Invitation.Executor/InvitationProcessTypeExecutor.cs

```
@@ -0,0 +1,139 @@
+        bool modified;
+        string? processMessage;
+
+        try
+        {
+            (nextStepTypeIds, stepStatusId, modified, processMessage) = processStepTypeId switch
+            {
+                ProcessStepTypeId.INVITATION_CREATE_CENTRAL_IDP => await _invitationProcessService.CreateCentralIdp(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_CREATE_SHARED_IDP_SERVICE_ACCOUNT => await _invitationProcessService.CreateSharedIdpServiceAccount(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_UPDATE_CENTRAL_IDP_URLS => await _invitationProcessService.UpdateCentralIdpUrl(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_CREATE_CENTRAL_IDP_ORG_MAPPER => await _invitationProcessService.CreateCentralIdpOrgMapper(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_CREATE_SHARED_REALM_IDP_CLIENT => await _invitationProcessService.CreateSharedIdpRealmIdpClient(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_ENABLE_CENTRAL_IDP => await _invitationProcessService.EnableCentralIdp(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_CREATE_DATABASE_IDP => await _invitationProcessService.CreateIdpDatabase(_companyInvitationId)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_CREATE_USER => await _invitationProcessService.CreateUser(_companyInvitationId, cancellationToken)
+                    .ConfigureAwait(false),
+                ProcessStepTypeId.INVITATION_SEND_MAIL => await _invitationProcessService.SendMail(_companyInvitationId)
+                    .ConfigureAwait(false),
+                _ => (null, ProcessStepStatusId.TODO, false, null)
```

there shouldn't be any valid processStepTypeIds left so default should rather throw UnexpectedConditionException to detect if we forgot any.

---

changed